tests/test_api.py

```diff
@@ -682,6 +683,34 @@ def test_component_tags_duplicate(client, api_path):
     assert response.json()["error"] == "Tag already exists."
 
 
+@pytest.mark.django_db(databases=("default", "read_only"), transaction=True)
+def test_product_stream_tag_filter(client, api_path):
+    ProductStreamFactory()
+    response = client.get(f"{api_path}/product_streams?tags=manifest")
+    assert response.status_code == 200
+    assert response.json()["count"] == 0
+
+    ProductStreamFactory(tag__name="manifest", tag__value="true")
+    response = client.get(f"{api_path}/product_streams?tags=manifest")
+    assert response.status_code == 200
+    assert response.json()["count"] == 1
+
+
+@pytest.mark.django_db(databases=("default", "read_only"), transaction=True)
+def test_product_stream_without_tag_filter(client, api_path):
+    ProductStreamFactory()
+    assert ProductStream.objects.exclude(tags__name="manifest").count() == 1
+    response = client.get(f"{api_path}/product_streams?tags=!manifest")
+    assert response.status_code == 200
+    assert response.json()["count"] == 1
+
+    ProductStreamFactory(tag__name="manifest", tag__value="true")
+    assert ProductStream.objects.exclude(tags__name="manifest").count() == 1
+    response = client.get(f"{api_path}/product_streams?tags=!manifest")
+    assert response.status_code == 200
+    assert response.json()["count"] == 1
```

Comment:
I think it's better to assert that `response[results][0]` is the stream we expect, instead of just checking the count:
```suggestion
    response = response.json()
    assert response["count"] == 1
    assert response["results"][0]["name"] == the_first_stream.name
```

Reply:
Good idea, thanks.